src/components/AssertionForm/useAssertionForm.ts

```diff
@@ -4,21 +4,32 @@ import {
   currenciesByChain,
   oov3AddressesByChainId,
 } from "@/constants";
-import { useMinimumBond } from "@/hooks";
-import type { ChainId, DropdownItem } from "@/types";
-import { useEffect, useState } from "react";
+import { truncateDecimalString } from "@/helpers";
+import { useBalanceAndAllowance, useMinimumBond } from "@/hooks";
+import type { ChainId, ChainName, DropdownItem } from "@/types";
+import { useEffect, useMemo, useState } from "react";
 import { useUpdateEffect } from "usehooks-ts";
 import type { Address } from "viem";
 import { formatUnits, parseUnits } from "viem";
 import { useAccount, useNetwork, useToken } from "wagmi";
 
 export type AssertionFormProps = ReturnType<typeof useAssertionForm>;
 
-export function useAssertionForm() {
-  const { address: userAddress, isConnected } = useAccount();
+function useChain() {
```

Comment:
the `useAssertionForm` hook was getting too big, so I split it out into some smaller ones that each have one responsibility